core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -193,13 +193,24 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
     endpoints.containsKey(name)
   }
 
+  def getNumOfThreads(conf: SparkConf, defaultNumThreads: Int): Int = {
+    val executorId = conf.get("spark.executor.id", "")
+    val isDriver = executorId == SparkContext.DRIVER_IDENTIFIER ||
+                    executorId == SparkContext.LEGACY_DRIVER_IDENTIFIER
+    val side = if (isDriver) "driver" else "executor"
+    val num = conf.getInt(s"spark.$side.rpc.netty.dispatcher.numThreads", defaultNumThreads)
+    if (num > 0) num else defaultNumThreads
+  }
+
   /** Thread pool used for dispatching messages. */
   private val threadpool: ThreadPoolExecutor = {
     val availableCores =
       if (numUsableCores > 0) numUsableCores else Runtime.getRuntime.availableProcessors()
     val numThreads = nettyEnv.conf.getInt("spark.rpc.netty.dispatcher.numThreads",
       math.max(2, availableCores))
-    val pool = ThreadUtils.newDaemonFixedThreadPool(numThreads, "dispatcher-event-loop")
+    val finalNumThreads = getNumOfThreads(nettyEnv.conf, numThreads)
```

Comment:
When driver or executor numThreads are set, we probably should not access the fallback option. Maybe refactor line 209-211?

Reply:
thanks chenzhao.
I think we still need the fallback because the value set by user might be invalid, like negative value.

Reply:
Yes, it means we need the fallback. The fallback value, numThreads, needs to be there.